### PR TITLE
STORY-14853: start building a Debian 11 base image

### PR DESCRIPTION
**Pull request by jebentier**
Debian 11 is starting to be required for multiple projects, so I'm going ahead and creating a separate tag for Debian 11 so we can systematically upgrade.

### CODE REVIEW

**Comment by brandonbell** [comment]
We'll either need to make the Dockerfile smarter or have a separate one for each version.  

Debian 11 would use the `bullseye` repositories instead of Stretch and we don't need to use the archives either.  

Should be similar to 
```
deb http://deb.debian.org/debian/ bullseye main 
deb http://security.debian.org/debian-security bullseye-security main 
```

**Comment by jebentier** [comment]
@brandonbell Do we need those additions for Debian 11 or can we not add them and force downstream to account for their own dependency needs?

File: Dockerfile [hunk 1]
```diff
@@ -1,4 +1,6 @@
-FROM debian:9
+ARG DEBIAN_VERSION=9
```
Comment (brandonbell) [inline Dockerfile:1]: Can we set this to `DEBIAN_VERSION=11` ?

**Review by brandonbell — COMMENTED** [review]
> @brandonbell Do we need those additions for Debian 11 or can we not add them and force downstream to account for their own dependency needs?

I updated my comment, but yes.  We only need to set the Apt sources to use the archives for Debian 9 and under.  

Can we also change the `Dockerfile.old` to something more descriptive, perhaps `Dockerfile.stretch`?

File: Dockerfile (hunk 1 above)
Comment (jebentier) [reply Dockerfile]: Done and Done 565f611be24432d3c24979c99146a7c735b58c88

**Review by brandonbell — APPROVED** [review]
LGTM

**Pull request merged**